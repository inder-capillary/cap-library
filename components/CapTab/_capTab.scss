@import '../styles/_variables.scss';

$capTab : cap-tab-v2;

.ant-tabs.#{$capTab} {
    .ant-tabs-nav .ant-tabs-tab-active {
        color: $FONT_COLOR_01;
    }
    
    .ant-tabs-tab-active.ant-tabs-tab.ant-tabs-tab-disabled{
        cursor: not-allowed;
        color: rgba(0, 0, 0, 0.25);
        font-weight:0;
        
    }
    
    .ant-tabs-tab.ant-tabs-tab-disabled:hover{
        cursor: not-allowed;
        color: rgba(0, 0, 0, 0.25) !important;
        font-weight:0;
        
    }
    .ant-tabs-nav .ant-tabs-tab:hover {
        color: $FONT_COLOR_01;
        background-color: #faf9f4;
    }
    
    
    .ant-tabs.ant-tabs-card .ant-tabs-card-bar .ant-tabs-tab-active {
        
        color: $FONT_COLOR_01;
    }
    
    .ant-tabs-ink-bar.ant-tabs-ink-bar-animated{
        background-color:rgba(0, 0, 0, 0.25);
    }
}




