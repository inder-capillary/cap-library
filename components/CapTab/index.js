/**
*
* CapTab
*
*/

import React from 'react';
import './_capTab.scss';
import PropTypes from 'prop-types';
import { Tabs } from "antd";
// import styled from 'styled-components';

const { TabPane } = Tabs;

class CapTab extends React.Component { // eslint-disable-line react/prefer-stateless-function
  constructor(props) {
    super(props);
    this.state = {
      activeKey: (props.panes ? `${props.panes[0].title}_0` : 0),
    };
  }

  onChange = (activeKey) => {
    this.setState({ activeKey });
  }

  render() {
    const {panes, ...rest} = this.props;
    return (
      <div className={this.props.className}>
        {panes ? (
          <Tabs
            animated
            onChange={this.onChange}
            activeKey={this.state.activeKey}
            {...rest}>
            { panes.map((pane, i) => <TabPane tab={pane.title} key={`${pane.title}_${i}`} disabled={pane.disabled} animated>{pane.content}</TabPane>)}
          </Tabs>
        ) : <div></div>}

      </div>
    );
  }
}

CapTab.propTypes = {
  panes: PropTypes.arrayOf(Object),
  className: PropTypes.string,
};

export default CapTab;
