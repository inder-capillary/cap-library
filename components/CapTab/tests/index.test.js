import renderer from 'react-test-renderer';
import React from 'react'; // eslint-disable-line no-unused-vars
import {mount} from 'enzyme';
import CapTab from '../index';

describe('<CapTab />', () => {
  const props = {
    panes: [{content: 'conatiner 1', title: 'tab 1'}],
  };
  it('CapTab renders correctly', () => {
    const tree = renderer.create(<CapTab {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('check the props is defined', () => {
    const CapTabComponent = mount(<CapTab {...props} />);
    expect(CapTabComponent.prop('panes')).toBeDefined();
  });

  it('render pane input correctly with object', () => {
    const CapTabComponent = mount(<CapTab {...props} />);
    expect(CapTabComponent.prop('panes')).toEqual([{content: 'conatiner 1', title: 'tab 1'}]);
  });
});
