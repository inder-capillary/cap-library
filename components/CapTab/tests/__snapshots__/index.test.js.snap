// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CapTab /> CapTab renders correctly 1`] = `
<div>
  <div />
</div>
`;
